packages/headless/doc-parser/src/actions-resolver.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  ApiEntryPoint,
+  ApiItemKind,
+  ApiVariable,
+} from '@microsoft/api-extractor-model';
+import {DocNode, DocParamBlock} from '@microsoft/tsdoc';
+import {AnyEntity, Entity} from './entity';
+import {emitAsTsDoc} from './tsdoc-emitter';
+
+interface ActionsNamespace {
+  name: string;
+  members: Action[];
+}
+
+enum ActionKind {
+  Action = 'action',
+  AsyncAction = 'asyncThunk',
+  AnalyticsAction = 'analyticsAction',
+}
+
+interface Action {
+  desc: string;
+  fullName: string;
+  parameter: AnyEntity | null;
+  actionKind: ActionKind;
+}
+
+export function resolveActionNamespaces(
+  entry: ApiEntryPoint
+): ActionsNamespace[] {
+  return entry.members
+    .filter(
+      (member) =>
+        member.kind === ApiItemKind.Namespace &&
+        member.displayName.endsWith('Actions')
+    )
+    .map((namespace) => ({
+      name: namespace.displayName,
+      members: namespace.members
+        .filter((member) => member.kind === ApiItemKind.Variable)
+        .map((action) => resolveAction(action as ApiVariable))
+        .filter((action) => action) as Action[],
+    }));
+}
+
+function resolveAction(variable: ApiVariable): Action | null {
+  const tokens = variable.excerpt.tokens.map(({text}) => text);
+
+  if (tokens[1] === 'AsyncThunkDefinition') {
+    const actionNameToken = tokens[2];
+    const [, actionName] = /"([^"]*)"/.exec(actionNameToken)!;
+    const parameterTypeName = tokens[4].trim() === ',' ? tokens[5] : null;
```

Comment:
In this PR, this is meant to check whether the parameter type is `void` since the api extractor doesn't identify `void` as being a reference and combines it with whatever non-reference tokens comes before and after it.

However, the api extractor doesn't recognize `string` or `number` as a reference either. This part would increase in complexity in the final code since it would need to be able to detect `string` and `number` parameters.

Reply:
Why should `void`, `string` and `number` be references instead of primitives? What would they reference?

Reply:
In the final JSON they would be expressed as primitives, what I meant is that unlike reference tokens they are mixed with normal tokens, so parsing them would require some extra RegExp and conditions.